src/accountant_skel.rs

```diff
@@ -245,8 +247,44 @@ impl<W: Write + Send + 'static> AccountantSkel<W> {
         }
         Ok(())
     }
+    /// Process verified blobs, already in order
+    /// Respond with a signed hash of the state
+    fn replicate_state(
+        obj: &Arc<Mutex<AccountantSkel<W>>>,
+        verified_receiver: &streamer::BlobReceiver,
+        blob_recycler: &packet::BlobRecycler,
+    ) -> Result<()> {
+        let timer = Duration::new(1, 0);
+        let blobs = verified_receiver.recv_timeout(timer)?;
+        for msgs in &blobs {
+            let blob = msgs.read().unwrap();
+            let entries: Vec<Entry> = deserialize(&blob.data()[..blob.meta.size]).unwrap();
+            for entry in entries {
+                obj.lock().unwrap().acc.register_entry_id(&entry.id);
+                for event in &entry.events {
```

Comment:
This loop is redundant at best, a security hole at worst. Delete the whole thing.

Reply:
entries don't get processed if this doesn't happen

Reply:
The line above this comment. You're already registering the correct entry ID in the correct place.

Reply:
If events are being rejected, that's because they contain a last_id that isn't one of the entry IDs in one of the previous 4096 entry IDs.

Reply:
No, the clients sync the ledger and use the most recent Entry ID they've observed to set the Transaction's last_id field. This let's the client be sure their transaction won't be played on a fork and allows the network to reject sufficiently old transactions. It also allows the network to throw away old signatures instead of accumulating them for eternity to guard against double-spend attacks.